cmd/clusterctl/client/alpha/rollout_restarter.go

```diff
@@ -53,3 +65,8 @@ func setRestartedAtAnnotation(proxy cluster.Proxy, name, namespace string) error
 	patch := client.RawPatch(types.MergePatchType, []byte(fmt.Sprintf("{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"cluster.x-k8s.io/restartedAt\":\"%v\"}}}}}", time.Now().Format(time.RFC3339))))
 	return patchMachineDeployment(proxy, name, namespace, patch)
 }
+
+func setRolloutAfter(proxy cluster.Proxy, name, namespace string) error {
+	patch := client.RawPatch(types.MergePatchType, []byte(fmt.Sprintf("{\"spec\":{\"rolloutAfter\":\"%v\"}}", time.Now().Format(time.RFC3339))))
```

Comment:
How does upstream handle this (kubectl's rollout restart mechanism for e.g. Deployments)?

Reply:
They're doing it similar (IIRC the `alpha rollout` subcommand was leaned on the upstream `kubectl rollout` subcommand)

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/rollout/rollout_restart.go#L189-L191

Reply:
Also for your question: https://github.com/kubernetes-sigs/cluster-api/pull/6858#discussion_r916224034 ? If so I'm fine with the solution :-)

(sorry to not answer on the old thread but it was gone due to changes on the files view...)